joomla-projects/gsoc22_guided-tour
# SQL fixes for PR #6

Pull Request for PR #6 .

### Summary of Changes

This Pull Request (PR) fixes errors in the installation and update SQL scripts and one error in the tours model and adapts the update SQL scripts to the enhancements made with PR https://github.com/joomla/joomla-cms/pull/37156 for the 4.2 version for more fault tolerant update SQL scripts.

The changes have been made with several commits in a comprehensive way so you can understand why a change has been made.

Simply go to the list of commits of this PR. You can expand the descriptions with details or links to details by using the "..." right beside the title of each commit.

By clicking on the title (or better right click and chose open in a new tab) you can then see the changes made with this commit.

### Testing Instructions

Check that a new installation and after that login to backend work without SQL errors on all kinds or supported databases (MySQL or MariaDB and PostgreSQL).

Check that updating any 3.10 or 4.x version to a package build with this PR and after that login to backend work without SQL errors on all kinds or supported databases (MySQL or MariaDB and PostgreSQL).

### Actual result BEFORE applying this Pull Request

New installations and updates fail for sure with PostgreSQL.

Hard-coded user IDs used for created by and modified by users for the new data on new installations and updates.

On PostgreSQL you get an SQL error due to hardcoded MySQL names quotes being used in an SQL query in the guided tours model.

MySQL data types were used in SQL scripts for PostgreSQL.

Wrong data types (strings instead of numbers or datetimes/timestamps) used for values in insert statements.

....

### Expected result AFTER applying this Pull Request

No such errors anymore.

### Documentation Changes Required

None.
